service/controller/resource/promxy/types.go

```diff
@@ -0,0 +1,208 @@
+package promxy
+
+import (
+	"fmt"
+	"time"
+
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/pkg/relabel"
+)
+
+type Promxy struct {
+	// Prometheus configs that includes configurations for
+	// recording rules, alerting rules, etc.
+	PromConfig config.Config `yaml:",inline"`
+	// Promxy specific configuration
+	Promxy PromxyConfig `yaml:"promxy"`
+}
+
+// PromxyConfig is the configuration for Promxy itself
+type PromxyConfig struct {
+	// Config for each of the server groups promxy is configured to aggregate
+	ServerGroups []*ServerGroup `yaml:"server_groups"`
+}
+
+func (p *PromxyConfig) contains(group *ServerGroup) bool {
+	for _, val := range p.ServerGroups {
+		if val.PathPrefix == group.PathPrefix {
+			return true
+		}
+	}
+	return false
+}
+
+func (p *PromxyConfig) add(group *ServerGroup) {
+	p.ServerGroups = append(p.ServerGroups, group)
+}
+
+func (p *PromxyConfig) remove(group *ServerGroup) {
+	var index int
+	for key, val := range p.ServerGroups {
+		if val.PathPrefix == group.PathPrefix {
+			index = key
+		}
+	}
+	p.ServerGroups = append(p.ServerGroups[:index], p.ServerGroups[index+1:]...)
+}
+
+// ServerGroup is the configuration for a ServerGroup that promxy will talk to.
+// This is where the vast majority of options exist.
+type ServerGroup struct {
```

Comment:
Would you be able to at least add a comment with some reasoning as to why we're using a struct copied over from promxy, instead of importing it? Just so that if we come back and want to re-visit this in a while we have some context of what to watch out for. I recon this will get out of sync sooner or later.

Reply:
I added it over the Promxy struct